oncall/oncall.go

```diff
@@ -168,6 +174,96 @@ func UpdateOnCallRotation(conf config.DeputizeConfig) error {
       return fmt.Errorf("Unable to add new users to LDAP")
     }
 
+  if conf.GitlabEnabled {
+    if !conf.Quiet {
+      log.Printf("Gitlab integration enabled. Gitlab group: %s", conf.GitlabGroup)
+    }
+
+    gitlabClient := gitlab.NewClient(nil, secret.Data["gitlabAuthToken"].(string))
+    gitlabClient.SetBaseURL(conf.GitlabServer+"api/v4")
+
+    // Lets get user ids for On Call people
+    for _, email := range newOnCallApproverEmails {
+      userOptions := &gitlab.ListUsersOptions{Search: gitlab.String(email)}
+      users, _, err := gitlabClient.Users.ListUsers(userOptions)
+      if err != nil {
+        log.Printf("Warning: Got %s back from Gitlab API\n", err)
+      } 
+      if len(users) == 1 {
+        // We expect only one user returned based on an email. We error out otherwise
+        if !conf.Quiet {
+          log.Printf("User found! username is %s for email %s\n", users[0].Username, email)
+        }
+        newOnCallApproverGitlabUserIds = append(newOnCallApproverGitlabUserIds, users[0].ID)
+      } else if len(users) == 0 {
+        if !conf.Quiet {
+          log.Printf("No user found for email %s\n", users[0].Username, email)
+        }        
+      } else {
+        // Lets output some helpful information if we don't get 1 user
+        if !conf.Quiet {
+          for _, user := range users {
+            log.Printf("Found the following users associated with \"%s\": %s\n", email, user.Username)
+          }
+        }
+        return fmt.Errorf("Found more than one user with an email of %s: %d users found", email, len(users))
+      }
+    }
+
+    if len(newOnCallApproverGitlabUserIds) == 0 {
+      // If no users are in the new approver list, leave the group alone
+      log.Printf("No new Approvers, not updating Gitlab group: %s", conf.GitlabGroup)
+    } else {
+      // Add OnCall approvers to approver group
+      if !conf.Quiet {
+        log.Printf("Updating Gitlab group: %s", conf.GitlabGroup)
+      }
+
+      // Remove existing members of the group, if they exist
+      if !conf.Quiet {
+        log.Printf("Removing old approvers %s from Gitlab group: %s", conf.GitlabGroup)
```

Comment:
missing two vars here

Reply:
Actually that was only supposed to have 1. Thanks for the catch!